src/main.ts

```diff
@@ -51,128 +52,105 @@ class RubyDebugSession extends DebugSession {
-			that.sendEvent(new OutputEvent(error.message, 'stderr'));
-		}).on('breakpointHit', (threadId: number) => {
-			that.sendEvent(new StoppedEvent('breakpoint', threadId));
-		}).on('suspended', (threadId: number) => {
-			that.sendEvent(new StoppedEvent('step', +threadId));
-		}).on('exception', (threadId: number, exceptionMsg: string) => {
-			that.sendEvent(new StoppedEvent('exception', threadId, exceptionMsg));
-		}).on('debuggerClientClose', () => {
-			that.sendEvent(new TerminatedEvent());
-		}).on('debuggerClientError', (msg: string) => {
-			that.sendEvent(new OutputEvent(msg));
-			that.sendEvent(new TerminatedEvent());
-		}).on('debuggerClientTimeout', (msg: string) => {
-			that.sendEvent(new OutputEvent(msg + '\n', 'stderr'));
-			that.sendEvent(new TerminatedEvent());
+			this.sendResponse(response);
+			this.sendEvent(new InitializedEvent());
+		}).on('debuggerComplete', () => {
+			this.sendEvent(new TerminatedEvent());
+		}).on('executableOutput', (data: Buffer) => {
+			this.sendEvent(new OutputEvent(data.toString(), 'stdout'));
+		}).on('executableStdErr', (error: Buffer) => {
+			this.sendEvent(new OutputEvent(error.toString(), 'stderr'));
+		}).on('nonTerminalError', (error: string) => {
+			this.sendEvent(new OutputEvent("Debugger error: "+error+ '\n', 'stderr'));
+		}).on('breakpoint', result => {
+			this.sendEvent(new StoppedEvent('breakpoint', result.threadId));
+		}).on('suspended', result => {
+			if ( args.stopOnEntry && !this._hasStopped )
+				this.sendEvent(new StoppedEvent('entry', result.threadId));
+			else
+				this.sendEvent(new StoppedEvent('step', result.threadId));
+			this._hasStopped = true;
+		}).on('exception', result => {
+			this.sendEvent(new OutputEvent("\nException raised: ["+result.type+"]: "+result.message+"\n",'stderr'));
+			this.sendEvent(new StoppedEvent('exception', result.threadId, result.type+": "+result.message));
+		}).on('terminalError', (error: string) => {
+			this.sendEvent(new OutputEvent("Debugger terminal error: "+ error))
+			this.sendEvent(new TerminatedEvent());
 		});
 
 		if (args.showDebuggerOutput) {
 			this.rubyProcess.on('debuggerOutput', (data: Buffer) => {
-				that.sendEvent(new OutputEvent(data.toString() + '', 'stderr'));
+				this.sendEvent(new OutputEvent(data.toString() + '\n', 'console'));
 			});
 		}
-
-		//this.sendResponse(response);
-
-		if (args.stopOnEntry) {
-			this.sendResponse(response);
-
-			// we stop on the first line
-			this.sendEvent(new StoppedEvent('entry', RubyDebugSession.THREAD_ID));
-		}
 	}
 
 	// Executed after all breakpints have been set by VS Code
-	protected configurationDoneRequest(response: DebugProtocol.ConfigurationDoneResponse, args:
+	protected configurationDoneRequest(response: DebugProtocol.ConfigurationDoneRequest, args:
 	DebugProtocol.ConfigurationDoneArguments): void {
-		var command = ['start'];
-		this.rubyProcess.Run(command.join(' ') + '\n');
+		this.rubyProcess.Run('start');
```

Comment:
It's `DebugProtocol. ConfigurationDoneResponse ` actually, and we need to send back the response.